packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/ParameterProvider.cs

```diff
@@ -175,5 +204,52 @@ private ParameterValidationType GetParameterValidation()
 
             return ParameterValidationType.AssertNotNull;
         }
+
+        private static ValueExpression? GetParameterDefaultValue(InputParameter inputParameter)
+        {
+            if (inputParameter.DefaultValue is null)
+            {
+                return null;
+            }
+
+            var defaultValue = inputParameter.DefaultValue.Value;
+            CSharpType valueType = CodeModelPlugin.Instance.TypeFactory.CreateCSharpType(inputParameter.DefaultValue.Type);
+
+            // handle default values for collections
+            if (defaultValue == null && valueType.IsCollection)
+            {
+                return New.Instance(valueType);
+            }
+            else if (valueType.IsFrameworkType && defaultValue is IConvertible)
+            {
+                // handle default values for framework types
+                var normalizedValue = Convert.ChangeType(defaultValue, valueType.FrameworkType);
+                return GetLiteralValue(normalizedValue);
+            }
+
+            return null;
+        }
+
+        private static ValueExpression? GetLiteralValue(object? value)
```

Comment:
feels like this belongs somewhere else, but not sure where.  It doesn't look like a unique implementation that would only apply to a parameterprovider

Reply:
I think we do not need this, `Snippet.Literal(value)` does exactly the same thing.

Reply:
`Snippet.Literal(value)` will throw which is why I avoided using it. Though since this should only ever be called for framework types, I think it should be okay to use it. Do let me know otherwise..